src/main/java/com/hyperwallet/clientsdk/util/HyperwalletEncryption.java

```diff
@@ -0,0 +1,182 @@
+package com.hyperwallet.clientsdk.util;
+
+import com.hyperwallet.clientsdk.HyperwalletException;
+import com.nimbusds.jose.Algorithm;
+import com.nimbusds.jose.EncryptionMethod;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWEAlgorithm;
+import com.nimbusds.jose.JWEHeader;
+import com.nimbusds.jose.JWEObject;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSObject;
+import com.nimbusds.jose.JWSSigner;
+import com.nimbusds.jose.Payload;
+import com.nimbusds.jose.crypto.RSADecrypter;
+import com.nimbusds.jose.crypto.RSAEncrypter;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jose.crypto.RSASSAVerifier;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.RSAKey;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.interfaces.RSAPublicKey;
+import java.text.ParseException;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+
+public class HyperwalletEncryption {
+
+    private static final String EXPIRATION = "exp";
+
+    private JWEAlgorithm encryptionAlgorithm = JWEAlgorithm.RSA_OAEP_256;
+    private JWSAlgorithm signAlgorithm = JWSAlgorithm.RS256;
+    private EncryptionMethod encryptionMethod = EncryptionMethod.A256CBC_HS512;
+    private final String clientPrivateKeySetLocation;
+    private final String hyperwalletKeySetLocation;
```

Comment:
can we expose `clientPrivateKeySetLocation` and `hyperwalletKeySetLocation` setters so that SDK user can fully use Builder pattern in case the client will do the following:
`GenericBuilder.of(HyperwalletEncryption::new).with(HyperwalletEncryption::setClientPrivateKeySetLocation, "").build();`

and don't forget to remove `final` keyword